### ISSUE TITLE
Rule disallowing top level becomes in playbooks

### ISSUE THREAD

**Issue by jpodivin** [ansible/ansible-lint#4287]
##### Summary

Become at the highest playbook level affects all executed tasks. Effectively setting become on all executed tasks.
This can cause issues with ownership of created files and directories, since these will all default to root. 

##### Issue Type

- new rule

**Comment by MatthiasKunnen** [comment]
My playbook starts with
```yaml
- name: Name here
  hosts: all
  become: true
  become_user: root
```

I then overwrite become_user on the task/block as needed. This approach allows me to set the user in a single place as this playbook is executed over ssh, locally, and with chroot. Otherwise, the user would be the connecting, executing, chrooted user, respectively.

Why would this be discouraged?

**Comment by jpodivin** [comment]
> My playbook starts with
> 
> ```yaml
> - name: Name here
>   hosts: all
>   become: true
>   become_user: root
> ```
> 
> I then overwrite become_user on the task/block as needed. This approach allows me to set the user in a single place as this playbook is executed over ssh, locally, and with chroot. Otherwise, the user would be the connecting, executing, chrooted user, respectively.
> 
> Why would this be discouraged?

I do agree that this is a good example of what top level become can be good for.

**Comment by jpodivin** [comment]
There is a considerable set of use cases this rule would effectivelly make more complicated.
